### ISSUE TITLE
A web UI for popular databases like MySQL, MongoDB, PostgreSQL

### ISSUE THREAD

**Issue by egoist** [egoist/ideas#3]
A web UI (like phpMyAdmin) for popular databases like MySQL, MongoDB, PostgreSQL etc, this would be very useful for local development, it could potentially use [TypeORM](https://typeorm.io/#/) to manage databases.

Example usage:
 
```bash
# Let's call it `dbadmin` for now
$ dbadmin
> The web UI is available at http://localhost:3456
```

**Comment by egoist** [comment]
The backend language you use to build this doesn't really matter, Rust? Probably better than Node.js but make sure you can find a good ORM.

**Comment by richardo2016** [comment]
[bookshelf](https://github.com/bookshelf/bookshelf) is recommended to manage SQL like databases.

Maybe it's nice to allow pass option to use http server which isn't driven by NodeJS, to manage database. That is, the UI just provide one interface to the http server rather than a specific runtime/library like NodeJS/TypeORM

**Comment by egoist** [comment]
@richardo2016 Yes, we can separate the UI and the API for database operations, the UI only needs to make REST or GraphQL calls to the API.

**Comment by daltondiaz** [comment]
I think the UI can be both a build in web and one in electronics, today I use DBeaver for many databases (MySQL, PostgreSQL and MSSQL) in a single IDE, this helps me a lot.

**Comment by djyde** [comment]
like phpmyadmin?

**Comment by daltondiaz** [comment]
@djyde yes, like phpmyadmin or like Hue [Hue demo ](http://demo.gethue.com/hue/editor/?editor=108020&type=impala).

**Comment by xialvjun** [comment]
have a look at https://github.com/KyleU/databaseflow

**Comment by 2nthony** [comment]
It would be better if also have desktop(electron) app

**Comment by unix** [comment]
@evillt It can be used in any envs if just a lightweight command-line tool. 
only run `$ dbadmin` or `$ npx dbadmin`, its look coool.

**Comment by kenshinji** [comment]
> The backend language you use to build this doesn't really matter, Rust? Probably better than Node.js but make sure you can find a good ORM.

@egoist Rust, ORM...hmm, you mean [this](https://github.com/diesel-rs/diesel)?

**Comment by nanne007** [comment]
I had come across the same idea one year ago.
There is a project https://github.com/nodeadmin/nodeadmin written in angular, but it seems not maintained anymore.
I had searched other oss solutions, and found no good one which can suit my need: 
- should be web ui.
- using user/pass of db to login in.
- rich interface of ddl like the nodeadmin mentioned  above.

Such project can be a killer of many desktop db managers like DBeaver.
